### PR TITLE
Find closest sub command

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following commit attempts to expose the ability to find the
closest subcommand that's on offer.

### CODE REVIEW

File: supercommand_test.go [hunk 1]
```diff
@@ -681,3 +681,25 @@ type flagAdderFunc func(*gnuflag.FlagSet)
...
+func (s *SuperCommandSuite) TestFindClosestSubCommandReturnsExactMatch(c *gc.C) {
+	sc := cmd.NewSuperCommand(cmd.SuperCommandParams{
+		UsagePrefix: "juju",
+		Name:        "command",
+		Log:         &cmd.Log{},
+	})
+	name, _, ok := sc.FindClosestSubCommand("help")
+	c.Assert(ok, gc.Equals, true)
+	c.Assert(name, gc.Equals, "help")
+}
```
Comment (mitechie) [inline supercommand_test.go]: can we have a test case for something that's off the wall and won't match please?

Comment (SimonRichardson) [reply supercommand_test.go]: of course...

**Comment by SimonRichardson** [comment]
@mitechie this is now done in https://github.com/juju/cmd/pull/65/commits/105e596d5271f0466a77ba80151cb3c1e32c7023

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**